game-app/game-core/src/main/java/games/strategy/triplea/ui/TabbedProductionPanel.java

```diff
@@ -46,95 +53,66 @@ static IntegerMap<ProductionRule> getProduction(
   @Override
   protected void initLayout() {
     this.removeAll();
-    this.setLayout(new GridBagLayout());
-    add(
-        new JLabel("Attack | Defense | Movement"),
-        new GridBagConstraints(
-            0,
-            0,
-            1,
-            1,
-            1,
-            1,
-            GridBagConstraints.EAST,
-            GridBagConstraints.HORIZONTAL,
-            new Insets(8, 8, 8, 0),
-            0,
-            0));
+    this.setLayout(new BorderLayout());
     final JTabbedPane tabs = new JTabbedPane();
-    add(
-        tabs,
-        new GridBagConstraints(
-            0,
-            1,
-            1,
-            1,
-            100,
-            100,
-            GridBagConstraints.EAST,
-            GridBagConstraints.BOTH,
-            new Insets(8, 8, 8, 8),
-            0,
-            0));
+    add(tabs, BorderLayout.CENTER);
     final ProductionTabsProperties properties =
-        ProductionTabsProperties.getInstance(gamePlayer, rules, uiContext.getResourceLoader());
+        new ProductionTabsProperties(gamePlayer, rules, uiContext.getResourceLoader());
     final List<Tuple<String, List<Rule>>> ruleLists = getRuleLists(properties);
+    ruleLists.removeIf(entry -> entry.getSecond().isEmpty());
+    cellSize = new Dimension();
+    for (Rule rule : rules) {
+      JPanel component = rule.getPanelComponent();
+      rulesComponents.put(rule, component);
+      cellSize.width = Math.max(cellSize.width, component.getPreferredSize().width);
+      cellSize.height = Math.max(cellSize.height, component.getPreferredSize().height);
+    }
```

Comment:
This block gets a bit dense, consider adding a guiding comment to help explain why we are picking max sizes. It's not immediately clear why one is better than the other, and what the overall goal is.